contracts/OwnedRegistrar.sol

```diff
@@ -50,22 +40,28 @@ contract OwnedRegistrar is RBAC {
     }
 
     /**
-     * @dev Migrate a name from the previous ENSRegistry and update its resolver.
+     * @dev Migrate a name from the previous ENSRegistry.
      * @param labelHash The hash of the label specifying the subnode.
      */
 
     function migrate(uint256 labelHash) public onlyRole("owner") {
         bytes32 node = keccak256(abi.encodePacked(baseNode, bytes32(labelHash)));
+
+        if(ens.owner(node) != address(0x0)) {
+            return;
+        }
+
         address owner = oldENS.owner(node);
+        address oldResolver = oldENS.resolver(node);
+
         ens.setSubnodeOwner(baseNode, bytes32(labelHash), address(this));
-        ens.setResolver(node, resolver);
-        Resolver(resolver).setAddr(node, owner);
+        ens.setResolver(node, oldResolver);
```

Comment:
With the new ENS registry contract, you can replace lines 57-59 with `ens.setSubnodeRecord`.